### PR TITLE
Add example code blocks in API documentation

### PR DESCRIPTION
**Pull request by aggarw13**
Add code blocks of API usage in documentation comments of API functions

### CODE REVIEW

File: source/include/jobs.h [hunk 1]
```diff
@@ -216,11 +250,96 @@ JobsStatus_t Jobs_GetTopic( char * buffer,
...
+ *     // AWS IoT Jobs service for the expected thing name.
+ *     // If the topic contains a jobID, then the pJobId and jobIdLength parameters
+ *     // populated by the API.
+ *
+ *     if( api == JobsJobsChanged )
+ *     { // Message from JobExecutionsChanged API.
+ *     }
+ *
+ *     else if( api == JobsNextJobChanged )
+ *     { // Message from  NextJobExecutionChanged API.
```
Comment (yourslab) [inline source/include/jobs.h:303]:
```suggestion
 *     { // Message from NextJobExecutionChanged API.
```

File: source/include/jobs.h [hunk 2]
```diff
@@ -339,7 +558,45 @@ JobsStatus_t Jobs_Describe( char * buffer,
...
+ * char topicBuffer[ JOBS_API_MAX_LENGTH( THING_NAME_LENGTH ) ] = { 0 };
+ * uint16_t topicLength = 0;
+ * JobsStatus_t status = JobsSuccess;
+ * status = Jobs_Update( topicBuffer,
+ *                         sizeof( topicBuffer ),
+ *                         THING_NAME,
+ *                         THING_NAME_LENGTH,
+ *                         pJobId,
+ *                         jobIdLength,
+ *                         &( topicLength ) );
```
Comment (yourslab) [inline source/include/jobs.h:591]: Indentation is off here

File: source/include/jobs.h [hunk 3]
```diff
@@ -339,7 +558,45 @@ JobsStatus_t Jobs_Describe( char * buffer,
...
+ *
+ * // The job ID is required to send a status update for a job to the AWS IoT
+ * // Jobs service.
+ * #define JOB_ID               "My_Job"
+ * #define JOB_ID_LENGTH        ( sizeof( JOB_ID ) - 1U )
+ *
+ * // This example allocated buffer of maximum length for a Jobs topic
+ * // possible for the thing name using the JOBS_API_MAX_LENGTH macro.
+ * char topicBuffer[ JOBS_API_MAX_LENGTH( THING_NAME_LENGTH ) ] = { 0 };
+ * uint16_t topicLength = 0;
```
Comment (yourslab) [inline source/include/jobs.h:583]:
```suggestion
 * uint16_t topicLength = 0U;
```

File: source/include/jobs.h [hunk 4]
```diff
@@ -339,7 +558,45 @@ JobsStatus_t Jobs_Describe( char * buffer,
...
+ * JobsStatus_t status = JobsSuccess;
+ * status = Jobs_Update( topicBuffer,
+ *                         sizeof( topicBuffer ),
+ *                         THING_NAME,
+ *                         THING_NAME_LENGTH,
+ *                         pJobId,
+ *                         jobIdLength,
```
Comment (yourslab) [inline source/include/jobs.h:590]: Should be using `JOB_ID` and JOB_ID_LENGTH`?

File: source/include/jobs.h [hunk 5]
```diff
@@ -308,7 +489,45 @@ JobsStatus_t Jobs_StartNext( char * buffer,
...
+ * // possible for the thing name using the JOBS_API_MAX_LENGTH macro.
+ * char topicBuffer[ JOBS_API_MAX_LENGTH( THING_NAME_LENGTH ) ] = { 0 };
+ * uint16_t topicLength = 0;
+ * JobsStatus_t status = JobsSuccess;
+ * status = Jobs_Describe( topicBuffer,
+ *                         sizeof( topicBuffer ),
+ *                         THING_NAME,
+ *                         THING_NAME_LENGTH,
+ *                         pJobId,
+ *                         jobIdLength,
```
Comment (yourslab) [inline source/include/jobs.h:521]: Should be using `JOB_ID` and `JOB_ID_LENGTH`?

File: source/include/jobs.h [hunk 6]
```diff
@@ -249,7 +368,38 @@ JobsStatus_t Jobs_MatchTopic( char * topic,
...
+ * // The following example shows usage of the Jobs_GetPending API to
+ * // generate topic string for the GetPendingJobExecutions API
+ * // of the AWS IoT Jobs service.
+ *
+ * // Every device should have a unique thing name registered with AWS IoT Core.
+ * // This example uses a dummy serial number for the thing name.
+ * #define THING_NAME           "11223445566"
+ * #define THING_NAME_LENGTH    ( sizeof( THING_NAME ) - 1U )
+ *
+ * // This example allocated buffer of maximum length for a Jobs topic
```
Comment (yourslab) [inline source/include/jobs.h:385]:
```suggestion
 * // This example allocates a buffer of maximum length for a Jobs topic
```

File: source/include/jobs.h (hunk 5 above)
Comment (aggarw13) [reply source/include/jobs.h]: Yes, fixed

File: source/include/jobs.h [hunk 7]
```diff
@@ -249,7 +368,38 @@ JobsStatus_t Jobs_MatchTopic( char * topic,
  * the buffer up to one less than the buffer size.  The topic is
  * ended with a NUL character.
  *
- * @note The thingName parameter does not need a NULL terminator.
+ * @note The thingName parameter does not need a NUL terminator.
```
Comment (yourslab) [inline source/include/jobs.h:371]: NULL-terminated string actually sounds like a misnomer considering that the proper label for the terminating character is NUL.

File: source/include/jobs.h (hunk 2 above)
Comment (aggarw13) [reply source/include/jobs.h:591]: Fixed

**Review by yourslab — APPROVED** [review]

File: source/include/jobs.h (hunk 7 above)
Comment (aggarw13) [reply source/include/jobs.h:371]: Interestingly, it is @dan4thewin who made me aware of the correct spellings for the `\0` character 😛

Comment (yourslab) [reply source/include/jobs.h:371]: haha same here

**Review by cobusve — APPROVED** [review]

**Pull request merged**